ray-curator/ray_curator/stages/classifiers/aegis.py

```diff
@@ -0,0 +1,555 @@
+            pretrained_model_name_or_path="meta-llama/LlamaGuard-7b",
+            padding_side="left",
+        )
+        self.tokenizer.pad_token = self.tokenizer.unk_token
+
+    def process_model_output(self, outputs: torch.Tensor) -> dict[str, np.ndarray]:
+        preds = outputs.cpu().numpy()
+        return {
+            "preds": preds,
+        }
+
+    def collect_outputs(self, processed_outputs: list[dict[str, np.ndarray]]) -> dict[str, np.ndarray]:
+        return {
+            "preds": np.concatenate([out["preds"] for out in processed_outputs], axis=0),
+        }
+
+    def create_output_dataframe(self, df_cpu: pd.DataFrame, collected_output: dict[str, np.ndarray]) -> pd.DataFrame:
+        df_cpu = df_cpu.drop(columns=["input_ids", "attention_mask"])
+
+        if self.add_instruction_data_guard:
+            df_cpu[self.prob_column] = collected_output["preds"].tolist()
+            df_cpu[self.pred_column] = (collected_output["preds"] >= 0.5).tolist()  # noqa: PLR2004
+        else:
+            df_cpu[self.pred_column] = collected_output["preds"].tolist()
+
+        return df_cpu
+
+    def process(self, batch: DocumentBatch) -> DocumentBatch:
+        processed_outputs = []
+        df_cpu = batch.to_pandas()
+
+        for model_input_batch in self.yield_next_batch(df_cpu):
+            # Forward pass
+            with torch.no_grad():
+                outputs = self.model(model_input_batch)
+
+            processed_output = self.process_model_output(outputs)
+            processed_outputs.append(processed_output)
+
+        # Collect all outputs
+        collected_output = self.collect_outputs(processed_outputs)
+
+        # Create output Pandas DataFrame
+        df_cpu = self.create_output_dataframe(df_cpu, collected_output)
+
+        # Sort by seq_order to preserve original order from tokenizer
+        if self.has_seq_order:
+            df_cpu = df_cpu.sort_values(by="_curator_seq_order", ignore_index=True).drop(
+                columns=["_curator_seq_order"]
+            )
+
+        return DocumentBatch(
+            task_id=batch.task_id,
+            dataset_name=batch.dataset_name,
+            data=df_cpu,
+            _metadata=batch._metadata,
+            _stage_perf=batch._stage_perf,
+        )
+
+
+@dataclass
+class WrapInPromptStage(ProcessingStage[DocumentBatch, DocumentBatch]):
+    """
+    WrapInPromptStage is a stage that truncates and wraps the input text in a prompt for the AEGIS model.
+    """
+
+    text_field: str
+    max_chars: int
+    _name = "wrap_in_prompt"
+
+    def inputs(self) -> tuple[list[str], list[str]]:
+        return ["data"], [self.text_field]
+
+    def outputs(self) -> tuple[list[str], list[str]]:
+        return ["data"], ["_hidden_text"]
+
+    def _wrap_in_prompt(self, df: pd.DataFrame) -> pd.DataFrame:
+        documents = df[self.text_field].tolist()
+        prompts = [format_aegis(doc[: self.max_chars]) for doc in documents]
+        df["_hidden_text"] = prompts
```

Comment:
instead of `_hiddent_text` what if we use `_curator_hidden_text` or something so that we don't override a column the user has coincidentally.. i understand unlikely but just to avoid namespace colision